VideoApp/Video-InternalTests/Specs/GridLayoutViewModelSpec.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  Copyright (C) 2022 Twilio, Inc.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Nimble
+import Quick
+@testable import VideoApp
+
+class GridLayoutViewModelSpec: QuickSpec {
+    override func spec() {
+        var sut: GridLayoutViewModel!
+        
+        beforeEach {
+            sut = GridLayoutViewModel()
+        }
+        
+        describe("addParticipant") {
+            context("when there are no pages") {
+                it("adds the participant to a new page") {
+                    sut.addParticipant(.stub(identity: "foo"))
+                    
+                    expect(sut.pages.count).to(equal(1))
```

Comment:
Unit tests should ideally have one expect statement per test.

Reply:
Yeah I'm with you that the fewer the asserts the better generally. And sometimes it works out that way like [this file](https://github.com/twilio/twilio-video-app-ios/blob/master/VideoApp/Video-InternalTests/Specs/PasscodeComponentsSpec.swift).

I took a stab at separating things out here but it was feeling like boilerplate that wasn't useful. 

I did some googling and the internet seemed to think the key is to test just one thing but is is ok and even preferable to use multiple asserts sometimes. These were the top search results:

1. https://stackoverflow.com/questions/762512/is-it-bad-practice-to-have-more-than-one-assertion-in-a-unit-test#:~:text=No%20it%20is%20not%20a,that%20should%20have%20been%20set.
2. https://softwareengineering.stackexchange.com/questions/7823/is-it-ok-to-have-multiple-asserts-in-a-single-unit-test
3. https://methodpoet.com/are-multiple-asserts-bad/
4. https://www.industriallogic.com/blog/multiple-asserts-are-ok/

So for this particular test I think adding the participant to a new page is essentially one thing and it's preferable to keep it simple. I could even create an extension that maps `Page` to a 2D array of identities like `[[String]]` and then it would be just one assert. But I'm not sure that makes the test better.

Note that further below for the `updateParticipant` tests, most of the time I have multiple tests per context because there really are several discrete things that should happen.

What do you think?

Reply:
Sounds reasonable.